dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/helpers.test.ts

```diff
@@ -0,0 +1,328 @@
+        {
+          correctionId: 1,
+          facilityId: 'FAC123',
+          exporter: 'Test Exporter',
+          additionalInfo: 'Pending correction 1',
+        },
+        {
+          correctionId: 3,
+          facilityId: 'FAC123',
+          exporter: 'Test Exporter',
+          additionalInfo: 'Pending correction 2',
+        },
+      ]);
+    });
+  });
+
+  describe('mapFeeRecordsToPendingCorrections', () => {
+    it('should return an empty array if there are no fee records', () => {
+      // Arrange
+      const feeRecords: FeeRecordEntity[] = [];
+
+      // Act
+      const result = mapFeeRecordsToPendingCorrections(feeRecords);
+
+      // Assert
+      expect(result).toEqual([]);
+    });
+
+    it(`should return an empty array if none of the fee records have status ${FEE_RECORD_STATUS.PENDING_CORRECTION}`, () => {
+      // Arrange
+      const statusesExludingPendingCorrection = Object.values(FEE_RECORD_STATUS).filter((status) => status !== FEE_RECORD_STATUS.PENDING_CORRECTION);
+      const feeRecords = statusesExludingPendingCorrection.map((status) =>
+        new FeeRecordEntityMockBuilder()
+          .withStatus(status)
+          .withCorrections([new FeeRecordCorrectionEntityMockBuilder().withIsCompleted(false).withAdditionalInfo('Pending correction 1').build()])
+          .build(),
+      );
+
+      // Act
+      const result = mapFeeRecordsToPendingCorrections(feeRecords);
+
+      // Assert
+      expect(result).toEqual([]);
+    });
+
+    it(`should return an empty array if none of the ${FEE_RECORD_STATUS.PENDING_CORRECTION} fee records have corrections`, () => {
+      // Arrange
+      const feeRecords = [
+        new FeeRecordEntityMockBuilder().withStatus(FEE_RECORD_STATUS.PENDING_CORRECTION).withCorrections([]).build(),
+        new FeeRecordEntityMockBuilder().withStatus(FEE_RECORD_STATUS.PENDING_CORRECTION).withCorrections([]).build(),
+      ];
+
+      // Act
+      const result = mapFeeRecordsToPendingCorrections(feeRecords);
+
+      // Assert
+      expect(result).toEqual([]);
+    });
+
+    it(`should return an empty array if all of the ${FEE_RECORD_STATUS.PENDING_CORRECTION} fee records' corrections are completed`, () => {
+      // Arrange
+      const feeRecords = [
+        new FeeRecordEntityMockBuilder()
+          .withStatus(FEE_RECORD_STATUS.PENDING_CORRECTION)
+          .withCorrections([new FeeRecordCorrectionEntityMockBuilder().withIsCompleted(true).build()])
+          .build(),
+        new FeeRecordEntityMockBuilder()
+          .withStatus(FEE_RECORD_STATUS.PENDING_CORRECTION)
+          .withCorrections([new FeeRecordCorrectionEntityMockBuilder().withIsCompleted(true).build()])
+          .build(),
+      ];
+
+      // Act
+      const result = mapFeeRecordsToPendingCorrections(feeRecords);
+
+      // Assert
+      expect(result).toEqual([]);
+    });
+
+    it(`should return an array of all of the pending corrections of the fee records`, () => {
```

Comment:
can more be added to this it (like the ones above) for why an array is being returned